### PR TITLE
[Optimization] Elemenwise fusion

### PR DESCRIPTION
**Pull request by BingooYang**
## Motivation

glm4.5模型性能优化

## Modifications

1. 添加cast+sgmoid + add融合算子（对应sglang torch compile自动融合部分）
2. 在glm模型中使能该融合算子
单测性能提升：
e37503fad40a7f8051262f1567d0e51b

## Usage or Command

no

## Accuracy Tests

fastdeploy/layers/fused_cast_sigmoid_bias.py

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [x] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: custom_ops/gpu_ops/fused_cast_sigmoid_bias.cu [hunk 1]
```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2025 PaddlePaddle Authors. All Rights Reserved.
```
Comment (heavengate) [inline custom_ops/gpu_ops/fused_cast_sigmoid_bias.cu]: 2025 -> 2026，下同

**Review by heavengate — COMMENTED** [review]
LGTM

File: custom_ops/gpu_ops/fused_cast_sigmoid_bias.cu [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    OutT* __restrict__ scores,
+    OutT* __restrict__ scores_with_bias,
+    const int num_experts) {
+  const int64_t token_idx = blockIdx.x;
+  const int64_t offset = token_idx * num_experts;
+
+  for (int j = threadIdx.x; j < num_experts; j += blockDim.x) {
+    float val = static_cast<float>(input[offset + j]);
+    // sigmoid: 1 / (1 + exp(-x))
+    float s = 1.0f / (1.0f + expf(-val));
```
[the remark this answers is not in the document]
Comment (BingooYang) [reply custom_ops/gpu_ops/fused_cast_sigmoid_bias.cu:60]: 当前场景不影响

File: fastdeploy/model_executor/layers/moe/fused_moe_marlin_backend.py [hunk 3]
```diff
@@ -260,6 +260,9 @@ def apply(
         if topk_method == "noaux_tc":
             from fastdeploy.model_executor.layers.moe.moe import get_moe_scores
 
+            use_fused = not layer.fd_config.load_config.dynamic_load_weight and current_platform.is_cuda()
```
Comment (zoooo0820) [inline fastdeploy/model_executor/layers/moe/fused_moe_marlin_backend.py]: 这个地方的限制条件可能不全面，需要确认下

**Review by zoooo0820 — DISMISSED** [review]

**Review by yongqiangma — DISMISSED** [review]
LGTM

File: tests/layers/test_fused_cast_sigmoid_bias.py [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+        print(
+            f"  tokens={num_tokens:5d}, experts={num_experts:3d} | "
+            f"ref={ref_time:8.1f}us, fused={fused_time:8.1f}us, speedup={speedup:.2f}x"
+        )
+
+    print()
+    print("  Note: The CUDA custom op fuses cast+sigmoid+bias into a single kernel,")
+    print("  eliminating 2 intermediate tensors and reducing kernel launches from 3 to 1.")
+    print("  Expected speedup: ~3x over the reference 3-op implementation.")
+    print("  Performance benchmark complete.\n")
```
[the remark this answers is not in the document]
Comment (freeliuzc) [reply tests/layers/test_fused_cast_sigmoid_bias.py:383]: 这里需要改小 iter，CI耗时确实会明显增加

**Review by freeliuzc — DISMISSED** [review]

File: fastdeploy/model_executor/layers/moe/moe.py [hunk 5]
```diff
@@ -91,13 +97,17 @@ def get_moe_scores(
...
     compute moe scores using e_score_correction_bias.
     """
-    scores = paddle.nn.functional.sigmoid(gating_output)
     assert e_score_correction_bias is not None, "e_score_correction_bias is none!"
-    scores_with_bias = scores + e_score_correction_bias
+    if use_fused_cast and current_platform.is_cuda():
+        scores, scores_with_bias = fused_cast_sigmoid_bias(gating_output, e_score_correction_bias, cast_type="float32")
+    else:
+        scores = paddle.nn.functional.sigmoid(gating_output)
+        scores_with_bias = scores + e_score_correction_bias
```
[the remark this answers is not in the document]
Comment (BingooYang) [reply fastdeploy/model_executor/layers/moe/moe.py:110]: fused_cast_sigmoid_bias不可用会在函数内部报错

**Review by yongqiangma — DISMISSED** [review]

**Review by freeliuzc — APPROVED** [review]

**Review by yongqiangma — DISMISSED** [review]

**Review by qingqing01 — DISMISSED** [review]

**Review by qingqing01 — APPROVED** [review]

**Review by yongqiangma — APPROVED** [review]
